src/formatting/pairs.rs

```diff
@@ -172,8 +184,11 @@ where
     RHS: Rewrite,
 {
```

Comment:
Inline suggestions below for moving the infix suffix/prefix spacing logic into this function as mentioned in the other inline comment in expr.rs. Written somewhat hastily inline in the GitHub Web UI (always dangerous :laughing:) so may not be perfect, but should be enough to express an idea how to implement the needed change